.github/workflows/dependencies/clang.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+#
+# Copyright 2024 The WarpX Community
+#
+# License: BSD-3-Clause-LBNL
+
+set -eu -o pipefail
+
+# `man apt.conf`:
+#   Number of retries to perform. If this is non-zero APT will retry
+#   failed files the given number of times.
+echo 'Acquire::Retries "3";' | sudo tee /etc/apt/apt.conf.d/80-retries
+
+# This dependency file is currently used within a docker container,
+# which does not come (among others) with wget, xz-utils, curl, git,
+# ccache, and pkg-config pre-installed.
+sudo apt-get -qq update
+sudo apt-get install -y \
+    cmake               \
+    libblas-dev         \
+    libc++-17-dev       \
+    libboost-math-dev   \
+    libfftw3-dev        \
+    libfftw3-mpi-dev    \
+    libhdf5-openmpi-dev \
+    liblapack-dev       \
+    libopenmpi-dev      \
+    libomp-17-dev       \
```

Comment:
Let's try to set `CXXFLAGS` so we use the LLVM  stdlib (libc++) and the LLVM OpenMP lib (libomp).

On Ubuntu, we think clang is configured to by default use the GNU libs for compatibility.